src/main/java/seedu/address/logic/parser/Parser.java

```diff
@@ -128,12 +129,24 @@ private Command prepareAdd(String args){
     private Command prepareDelete(String args) {
 
         Optional<Integer> index = parseIndex(args);
+        // check if argument is index
+
         if(!index.isPresent()){
-            return new IncorrectCommand(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE));
+            String nameMatch = args.length() > 0 ? args.substring(1) : args;
```

Comment:
i didnt? prepareDelete(arg) takes an argument, this can be both index and task name. processing is done in the code body. Im not sure why you're confused here.